keyvault/client/src/main/java/com/azure/keyvault/KeyvaultCredentialsPolicy.java

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault;
+
+import com.azure.common.credentials.ServiceClientCredentials;
+import com.azure.common.http.HttpPipelineCallContext;
+import com.azure.common.http.HttpResponse;
+import com.azure.common.http.HttpPipelineNextPolicy;
+import com.azure.common.http.policy.HttpPipelinePolicy;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Objects;
+
+final class KeyvaultCredentialsPolicy implements HttpPipelinePolicy {
+
+    private static final String HOST_HEADER = "Host";
+    private static final String AUTHORIZATION_HEADER = "Authorization";
```

Comment:
I don't think we need this credentials policy at all. It is not doing anything in regards to auth that is special and could be replaced with: [AsyncCredentialsPolicy.java](https://github.com/Azure/azure-sdk-for-java/blob/master/common/azure-common/src/main/java/com/azure/common/http/policy/AsyncCredentialsPolicy.java#L18)

Reply:
CredentialsPolicy and AsyncCredentialsPolicy in azure common class are generic, 
In bearer case we need to prefix the authorization header with “Bearer”

Plus,
In KV secrets API currently credentials class is not defined in the API , and is currently expected to be user implemented via ServiceCredentials interface or AsyncServiceCredentials interface. 
The credentials class should not be expected to add the “Bearer” prefix either.

Currently the custom KeyvaultCredentialsPolicy in secrets API is adding the Bearer prefix. If we use the generic credential policy as requested in review, I propose we  build a custom package private credentials wrapper which adds the “Bearer” prefix to the auth value/token generated.

Reply:
Introduced TokenCredentials and AsyncTokenCredentials classes.
AsyncTokenCredentails doesn't exist in Azure-common yet, created it internally for now.
Once auth logic is finalized by Scott Schaab, we will revisit this and decide the final approach.